sn_transfers/src/wallet/hot_wallet.rs

```diff
@@ -200,12 +268,11 @@ impl HotWallet {
     }
 
     /// Moves all files for the current wallet, including keys and cashnotes
-    /// to directory root_dir/wallet_ADDRESS
+    /// to directory root_dir/wallet_DATETIME
     pub fn stash(root_dir: &Path) -> Result<PathBuf> {
-        let wallet = HotWallet::load_from(root_dir)?;
         let wallet_dir = root_dir.join(WALLET_DIR_NAME);
-        let addr_hex = &format!("{:?}", wallet.address());
-        let new_name = format!("{WALLET_DIR_NAME}_{addr_hex}");
+        let datetime_str = Local::now().format("%Y-%m-%d_%H-%M-%S").to_string();
+        let new_name = format!("{WALLET_DIR_NAME}_{datetime_str}");
         let moved_dir = root_dir.join(new_name);
```

Comment:
Why switch to date/time?

Reply:
Makes it a lot easier to stash encrypted wallets without needing to decrypt them to get the public address.

Reply:
> without needing to decrypt them to get the public address.

Not sure I understand, how can you get the public address if it's not in the name anymore?

Reply:
Oooh I see! Although sorting by key is better, it's a hack so we don't need to unlock the wallet so do that?
But then on the other hand, filtering by date here is weird, and we will need to cleanup old ones every time...
We will also need to filter by public key when we want to support multi-key in the future so keeping pks is cleaner.
How about keeping the public key in a non-encrypted file along with the wallet?

Reply:
> Oooh I see! Although sorting by key is better, it's a hack so we don't need to unlock the wallet so do that?

Correct!

>  How about keeping the public key in a non-encrypted file along with the wallet?

I thought about using the existing `main_pubkey` file. But the drawback for me was that we can not verify whether that pubkey is actually valid for a locked encrypted wallet/secret key. We'd still have to unlock the wallet to verify if the pubkey actually belongs to that wallet.

> We will also need to filter by public key when we want to support multi-key in the future so keeping pks is cleaner.

When the project starts supporting multiple keys/wallets, I agree that wallets should be saved in folder names that represent their pubkey. I'm not sure what folder structure will be used then (perhaps https://github.com/maidsafe/safe_network/issues/1944 ?). In that case, I see the current `stash` more as a function to backup an old wallet that you may or may not know the password of.

If we do decide on stashing old wallets under a folder name that contains the public address. I'm just a bit stuck on figuring out how we would best deal with stashing encrypted wallets where the user doesn't remember the password of?

Reply:
Hey @grumbach ,

I've made the changes we've discussed (using the `main_pubkey` file to get the wallet address): https://github.com/maidsafe/safe_network/pull/1949/commits/ce62fbf7f7da0a9687e61579879c38ba75529c5a